components/tikv_alloc/src/lib.rs

```diff
@@ -130,6 +137,29 @@ mod imp {
         String::from_utf8_lossy(&buf).into_owned()
     }
 
+    #[derive(Default)]
+    pub struct JemallocStats {
+        pub allocated: usize,
+        pub active: usize,
+        pub metadata: usize,
+        pub resident: usize,
+        pub mapped: usize,
+        pub retained: usize,
+    }
+
+    pub fn fetch_stats() -> Result<JemallocStats, i32> {
+        let mut stats: JemallocStats = Default::default();
+        unsafe {
+            jemallocator::mallctl_fetch(STATS_ALLOCATED, &mut stats.allocated)?;
```

Comment:
jemalloc_ctl crate claim to be more efficient than `jemallocator::mallctl_fetch`. I plan to change to that if the PR look ok overall.
https://docs.rs/jemalloc-ctl/0.2.0/jemalloc_ctl/